json-path/src/main/java/com/jayway/jsonpath/internal/JsonContext.java

```diff
@@ -143,9 +146,13 @@ public Object readRoot(String[] paths, Predicate... filters) {
             	}
             }
         }
+        if (ret == null) {
+        	// Return an empty map.
+        	ret = configuration.jsonProvider().createMap();
+        }
```

Comment:
Why do you need this?

Reply:
To make sure we are not returning null map when, processing a jsonpath using readroot, throws an exception.